app/models/section.rb

```diff
@@ -4,15 +4,22 @@ class Section < ActiveRecord::Base
   validates  :crn, presence: true, uniqueness: true
   default_scope { order(name: :asc) }
   before_save :sort_periods, if: :periods_changed?
+  after_save :update_conflicts!, if: :periods_changed?
+
+  def self.compute_conflict_ids_for id
+    find_by_sql("SELECT sections.id FROM sections WHERE sections.id IN
+      (SELECT(unnest(conflict_ids(#{id.to_i})))) ORDER BY ID").map(&:id)
+  end
 
   def conflicts_with(section)
+    # TODO: should check the list of conflicts first
     i = 0
     while i < num_periods
       j = 0
       while j < section.num_periods
         if (periods_day[i] == section.periods_day[j] \
-          && ((periods_start[i] <= section.periods_start[j] && periods_end[i] >= section.periods_start[j]) \
-          || (periods_start[i] >= section.periods_start[j] && periods_start[i] <= section.periods_end[j])))
+          && ((periods_start[i].to_i <= section.periods_start[j] && periods_end[i].to_i >= section.periods_start[j]) \
```

Comment:
As I said (recording it here for posterity): the periods_start and periods_end fields are String[], not int[], and this function doesn't assume that they will always be a correct 4-digit military time, so for example "800" would have appeared to be _after_ "1000", which is wrong. Converting to int removes this ambiguity and that has been here for several months (22e731b).